libraries/AP_Baro/AP_Baro_UAVCAN.cpp

```diff
@@ -0,0 +1,77 @@
+#include <AP_HAL/AP_HAL.h>
+
+#if HAL_WITH_UAVCAN
+
+#include "AP_Baro_UAVCAN.h"
+#include <AP_BoardConfig/AP_BoardConfig.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <drivers/drv_baro.h>
+#include <drivers/drv_hrt.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define debug_baro_uavcan(level, fmt, args...) do { if ((level) <= AP_BoardConfig::get_can_debug()) { hal.console->printf(fmt, ##args); }} while (0)
+
+// There is limitation to use only one UAVCAN barometer now.
+
+/*
+  constructor - registers instance at top Baro driver
+ */
+AP_Baro_UAVCAN::AP_Baro_UAVCAN(AP_Baro &baro) :
+    AP_Baro_Backend(baro)
+{
+    _instance = _frontend.register_sensor();
+    if (hal.can_mgr != nullptr) {
+        AP_UAVCAN *ap_uavcan = hal.can_mgr->get_UAVCAN();
+        if (ap_uavcan != nullptr) {
+            // Give time to receive some packets from CAN if baro sensor is present
+            // This way it will get calibrated correctly
+            hal.scheduler->delay(2000);
```

Comment:
Why isn't a delay when initiating UAVCAN enough (like we had before)? As UAVCAN gets more drivers this will increase the overall boot time.